dali/kernels/audio/mel_scale/mel_filter_bank_cpu.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_AUDIO_MEL_SCALE_MEL_FILTER_BANK_CPU_H_
+#define DALI_KERNELS_AUDIO_MEL_SCALE_MEL_FILTER_BANK_CPU_H_
+
+#include <memory>
+#include "dali/core/common.h"
+#include "dali/core/error_handling.h"
+#include "dali/core/format.h"
+#include "dali/core/util.h"
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/audio/mel_scale/mel_filter_bank_args.h"
+
+namespace dali {
+namespace kernels {
+namespace audio {
+
+template <typename T = float, int Dims = 2>
+class DLL_PUBLIC MelFilterBankCpu {
+ public:
+  static_assert(std::is_floating_point<T>::value, "Only floating point types are supported");
+  static_assert(Dims >= 2, "At least 2 dimensions are expected");
+
+  DLL_PUBLIC ~MelFilterBankCpu();
+
+  DLL_PUBLIC KernelRequirements Setup(KernelContext &context,
+                                      const InTensorCPU<T, Dims> &in,
+                                      const MelFilterBankArgs &args);
+
+  DLL_PUBLIC void Run(KernelContext &context,
+                      const OutTensorCPU<T, Dims> &out,
+                      const InTensorCPU<T, Dims> &in,
+                      const MelFilterBankArgs &args);
+
+ private:
+  class Impl;
+  std::shared_ptr<Impl> impl_;
```

Comment:
In this case, please delete or implement copy and assignment.

Reply:
Just discovered - you just need out-of-place default constructor to use unique_ptr here. Will do the same in my PR.

Reply:
Applied, thanks!